api/scenario_api_simulator/include/scenario_api_simulator/npc_route_manager.h

```diff
@@ -17,41 +17,35 @@
 #ifndef NPC_ROUTE_MANAGER_H_INCLUDED
 #define NPC_ROUTE_MANAGER_H_INCLUDED
 
-#include <geometry_msgs/PoseStamped.h>
-#include <geometry_msgs/PoseWithCovarianceStamped.h>
-#include <geometry_msgs/TransformStamped.h>
-#include <geometry_msgs/TwistStamped.h>
+#include <autoware_lanelet2_msgs/msg/map_bin.hpp>
+#include <geometry_msgs/msg/pose.hpp>
 #include <lanelet2_core/LaneletMap.h>
-#include <lanelet2_core/geometry/BoundingBox.h>
 #include <lanelet2_core/geometry/Lanelet.h>
-#include <lanelet2_core/geometry/Point.h>
-#include <lanelet2_extension/utility/message_conversion.h>
 #include <lanelet2_extension/utility/utilities.h>
-#include <lanelet2_routing/RoutingGraph.h>
-#include <lanelet2_traffic_rules/TrafficRulesFactory.h>
-#include <npc_simulator/Object.h>
-#include <ros/ros.h>
+#include <npc_simulator/msg/object.hpp>
+#include <rclcpp/rclcpp.hpp>
 
+#include <string>
 #include <unordered_map>
+#include <vector>
 
-#include <scenario_api_utils/scenario_api_utils.h>
 
 namespace lanelet
 {
-class Lanelet;
-class LaneletMap;
-using LaneletMapPtr = std::shared_ptr<LaneletMap>;
-namespace routing
-{
-class RoutingGraph;
-}
-namespace traffic_rules
-{
-class TrafficRules;
-}
+  class Lanelet;
+  class LaneletMap;
+  using LaneletMapPtr = std::shared_ptr < LaneletMap >;
+  namespace routing
+  {
+    class RoutingGraph;
+  }
+  namespace traffic_rules
+  {
+    class TrafficRules;
+  }
 }  // namespace lanelet
 
-class NPCRouteManager
+class NPCRouteManager: public rclcpp::Node
```

Comment:
Similar to ScenarioAPISimulator class, this should instead take rclcpp::Node::SharedPtr as an input to the constructor and use it for setting parameters and pub/subs.